### PR TITLE
CI: unvendor behave

### PR DESCRIPTION
**Pull request by jchampio**
The main point of this PR is to allow developers to run the behave tests using the version of behave that is installed on their local machine, instead of vendoring this (developer-only) dependency through a tarball. Updating that dependency has taken a lot of effort recently, we don't see many benefits with the current scheme, and we'd like to simplify our Python dependency thicket in general. This is our first step.

Every added piece of complexity here, then, is to work around problems with our tight coupling to the vendored Python installation in the CI. The additional debt here is intended to be temporary, as we solve those problems in the vendored Python and move towards our eventual end goal: make a vendored Python stack totally optional for every user. (We'll start a gpdb-dev@ thread on our vendored-Python findings.)

The first commit removes the vendored `behave` from our Makefile setup, introduces a `pip`-installable `requirements-dev.txt` file to document the minimum required version, and performs a `pip install` on the CI jobs using that requirements file.
- This `requirements-dev.txt` is separate from the top-level `python-dependencies*` files on purpose, to start fresh. We plan to consolidate this little-by-little as we continue to reduce our dependency on vendored Python libraries.
- This commit focuses on CentOS6, and on that platform, Python 2.6 is installed by default (which our utilities do not support). We rely on `virtualenv` to set up an environment that is totally separate from the system Python.

The second commit works around an issue with our vendored Python installation: you can't run the vendored `python` executable without also setting `LD_LIBRARY_PATH`. On systems that ship Python 2.7 by default, setting `LD_LIBRARY_PATH` effectively breaks the system Python, making `virtualenv` unusable. We work around the problem by using `patchelf` to set up an `RPATH`/`RUNPATH` on the vendored Python, which is something that we likely want to do in the future for the official build.
- This works for CentOS6, CentOS7, and SLES12 -- SLES11 doesn't have a `patchelf` in its distribution, but we also don't run any tests there currently, so we've chosen not to worry about that until we need to.

**TL;DR: Debt comes out of the Makefiles that affect everyone and into the CI machines where it's isolated, to slowly simplify our dependency tree.**

## Reminders
- [x] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by jchampio** [comment]
@kmacoskey @bradfordboyle You may be interested in some of this.

File: concourse/scripts/behave_gpdb.bash [hunk 1]
```diff
@@ -5,24 +5,85 @@ set -eox pipefail
...
+		BEHAVE_FLAGS="${BEHAVE_FLAGS}"
 		if [ ! -z "\${BEHAVE_TAGS}" ]; then
 		    make -f Makefile.behave behave tags=\${BEHAVE_TAGS}
 		else
 		    flags="\${BEHAVE_FLAGS}" make -f Makefile.behave behave
 		fi
 	EOF
 
-	chmod a+x /opt/run_test.sh
+    chmod a+x /opt/run_test.sh
```
Comment (bradfordboyle) [inline concourse/scripts/behave_gpdb.bash]: Nitpick: The diff looks like it's adding a mixture of spaces and tabs.

File: concourse/scripts/behave_gpdb.bash [hunk 2]
```diff
@@ -5,24 +5,85 @@ set -eox pipefail
...
 source "${CWDIR}/common.bash"
 
-function gen_env(){
-  cat > /opt/run_test.sh <<-EOF
+function install_python_hacks() {
+    # Our Python installation doesn't run standalone; it requires
+    # LD_LIBRARY_PATH which causes virtualenv to fail (because the system and
+    # vendored libpythons collide). We'll try our best to install patchelf to
+    # fix this later, but it's not available on all platforms.
+    if which zypper > /dev/null; then
```
Comment (bradfordboyle) [inline concourse/scripts/behave_gpdb.bash]: FYI: `which` is not part of the POSIX specification while `command` is.

**Comment by bradfordboyle** [comment]
I don't think I have enough Python context to fully comment on the technical content. There is some "DRY"-ing that could be done, but as you point out, the added technical debt should be short lived. This change looks to be isolated to running `behave` tests, so I see no issue w/ merging it.

File: concourse/scripts/behave_gpdb.bash (hunk 1 above)
Comment (jchampio) [reply concourse/scripts/behave_gpdb.bash]: By design, unfortunately... `<<-EOF` only ignores leading tabs, not leading spaces. (Ctrl-F for "Here Documents" at the [man page](https://linux.die.net/man/1/bash)).

File: concourse/scripts/behave_gpdb.bash (hunk 2 above)
Comment (jchampio) [reply concourse/scripts/behave_gpdb.bash]: Do we have platforms that provide only the POSIX minimums? We haven't been focusing on strict POSIXisation; do we need to?

**Comment by jchampio** [comment]
@bradfordboyle Thanks for the review!

**Comment by jchampio** [comment]
Pushed a fixup to also regenerate the master pipeline from the template changes. That'll need to be rebased and regenerated once the PR is approved.

**Comment by marksliva** [comment]
lgtm. Something we should keep in mind for future work on some of these scripts is to rename them when it makes sense, i.e. rename behave_gpdb.bash to behave_gpdb_demo_cluster.bash

File: concourse/scripts/behave_gpdb.bash (hunk 2 above)
Comment (jmcatamney) [reply concourse/scripts/behave_gpdb.bash]: The utilities broadly assume Linux environments rather than pure POSIX, and gpinitsegment already uses `which` to find binaries, so using `which` is fine here.
